### DISCUSSION TITLE
Create a function node that may have multiple lines

### DISCUSSION THREAD

**Discussion by foodornt** [L3MON4D3/LuaSnip#962] [Q&A] [answered]
Just trying to add a new line causes luasnip to fail:
```lua
return {
  s(
    { trig = "H(%d)", regTrig = true },
    f(function(args, snip)
      return string.rep("\n", tonumber(snip.captures[1]))
    end)
  ),
}

```
and I don't want to use dynamic node as it needs another keypress
I guess I need an fmt node, although I can't return a node from a function node
any ideas on how to fix?

**Comment by leiserfg** [comment, marked as answer]
You should return a list with one element per line instead.

**Reply by foodornt** [reply]
i.e. this way?
```lua
return {
  s(
    { trig = "H(%d)", regTrig = true },
    f(function(args, snip)
      local list = {}
      for x = 1, tonumber(snip.captures[1]) do
        table.insert(list, "\n")
      end
      return list
    end)
  ),
}
```
still doesn't work

**Reply by foodornt** [reply]
DUMB me, it's just an empty string instead of "\n"
Thanks!
